map/src/menu/tracks/CloudTrackItem.jsx

```diff
@@ -100,26 +144,69 @@ export default function CloudTrackItem({ file, customIcon = null }) {
         }
     }
 
-    const rendered = useMemo(() => {
+    useEffect(() => {
+        if (ctx.openedPopper && ctx.openedPopper !== anchorEl) {
+            setOpenActions(false);
+        }
+    }, [ctx.openedPopper]);
+
+    return useMemo(() => {
         const trackName = TracksManager.getFileName(file);
         return (
             <>
                 <Tooltip title={info} arrow placement={mobile ? 'bottom' : 'right'} disableInteractive>
-                    <MenuItem id={'se-cloud-track-' + trackName} onClick={() => setDisplayTrack(true)}>
-                        <ListItemText inset>
-                            <Typography variant="inherit" noWrap>
-                                {customIcon}
+                    <MenuItem
+                        className={styles.item}
+                        divider
+                        id={'se-cloud-track-' + trackName}
+                        onClick={() => setDisplayTrack(true)}
+                    >
+                        <ListItemIcon className={styles.icon}>
+                            <TrackIcon />
+                        </ListItemIcon>
+                        <ListItemText>
+                            <Typography variant="inherit" className={styles.groupName} noWrap>
                                 {trackName}
                             </Typography>
+                            <Typography variant="body2" className={styles.groupInfo} noWrap>
+                                {dist && `${dist} km · `}
+                                {time && `${time}`}
+                                {points && ` · ${points}`}
```

Comment:
Please use better validation for real values, eg dist > 0, points > 0, etc (think about time :-)